src/actors/objects/http.rs

```diff
@@ -32,17 +51,17 @@ pub(super) fn prepare_downloads(
     .map(|download_path| FileId::Http(source.clone(), download_path))
     .collect();
 
-    Box::new(Ok(ids).into_future())
+    Box::new(future::ok(ids))
 }
 
 pub(super) fn download_from_source(
     source: Arc<HttpSourceConfig>,
     download_path: &DownloadPath,
 ) -> Box<dyn Future<Item = Option<DownloadStream>, Error = ObjectError>> {
-    // XXX: Probably should send an error if the URL turns out to be invalid
-    let download_url = match source.url.join(&download_path.0) {
+    // This can effectively never error since the URL is always validated to be a base URL.
+    let download_url = match join_url_encoded(&source.url, &download_path.0) {
         Ok(x) => x,
-        Err(_) => return Box::new(Ok(None).into_future()),
+        Err(_) => return Box::new(future::ok(None)),
```

Comment:
is this something worth logging? so we catch any invalid things that might flow through here sometime in the future?

Reply:
As the comment above states, this is something that "cannot happen" ™️  since the base URL is validated beforehand already. I just don't like to write `.unwrap()`.

As for the additional tests: Could add, but we already test the space, which shows that percent encoding works. I'm not sure if we should be unit-testing the `url` crate.